### ISSUE TITLE
bug(Reddit - Hide ads): `Hide ads` patch fails on `2024.17.0`

### ISSUE THREAD

**Issue by KobeW50** [inotia00/ReVanced_Extended#2592]
### Type

Error while patching

### Tools used

RVX Manager

### Application

Reddit 2024.17.0

RVX Manager 1.23.5

### Bug description

The fingerprint for the Hide ads patch fails to be resolved on Reddit 2024.17.0. 

### Error logs

```shell
- Device Info
RVX Manager: 1.23.5
Model: SM-S906U1
Android version: 14
Supported architectures: arm64-v8a, armeabi-v7a, armeabi
Root permissions: No

- Patch Info
App: com.reddit.frontpage v2024.17.0 (Suggested: Any)
Patches version: v5.1.1
Patches added: Default
Patches removed: None
Default patch options changed: None

- Settings
Allow changing patch selection: true
Version compatibility check: true
Show universal patches: true
Remove unused library: true
Use pre-release: false
Patches source: inotia00/revanced-patches

- Logs
Reading APK
Decoding app manifest
Loading patches
Deleting existing temporary files directory
Decoding resources
Initializing lookup maps
Executing patches
Applied 9 patches
Hide ads failed: app.revanced.patcher.patch.PatchException: Failed to resolve newAdPostFingerprint
	at app.revanced.util.fingerprint.LegacyFingerprintKt.getException(LegacyFingerprint.kt:24)
	at app.revanced.util.fingerprint.LegacyFingerprintKt.methodOrThrow(LegacyFingerprint.kt:64)
	at app.revanced.patches.reddit.ad.AdsPatchKt.adsPatch$lambda$14$lambda$13(AdsPatch.kt:116)
	at app.revanced.patches.reddit.ad.AdsPatchKt.$r8$lambda$dd1evXuU7Elr_lrTZzwxrZL-q80(Unknown Source:0)
	at app.revanced.patches.reddit.ad.AdsPatchKt$$ExternalSyntheticLambda5.invoke(Unknown Source:2)
	at app.revanced.patcher.patch.Patch.execute(SourceFile:71)
	at app.revanced.patcher.patch.BytecodePatch.execute$revanced_patcher(SourceFile:166)
	at app.revanced.patcher.Patcher$invoke$1.invokeSuspend$execute(SourceFile:84)
	at app.revanced.patcher.Patcher$invoke$1.invokeSuspend(SourceFile:109)
	at app.revanced.patcher.Patcher$invoke$1.invoke(SourceFile:0)
	at app.revanced.patcher.Patcher$invoke$1.invoke(SourceFile:0)
	at kotlinx.coroutines.flow.SafeFlow.collectSafely(SourceFile:57)
	at kotlinx.coroutines.flow.AbstractFlow.collect(SourceFile:226)
	at app.revanced.manager.flutter.MainActivity$runPatcher$1$patcherResult$1$1.invokeSuspend(SourceFile:323)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(SourceFile:33)
	at kotlinx.coroutines.DispatchedTask.run(SourceFile:101)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(SourceFile:263)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(SourceFile:95)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(SourceFile:69)
	at kotlinx.coroutines.BuildersKt.runBlocking(SourceFile:1)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(SourceFile:47)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(SourceFile:1)
	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda$34(SourceFile:317)
	at app.revanced.manager.flutter.MainActivity.$r8$lambda$Oku38Rh6VOg9CiulxEElyw3q4GI(SourceFile:0)
	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda0.run(SourceFile:0)
	at java.lang.Thread.run(Thread.java:1012)
Compiling modified resources
Ripping selected native libs
Aligning APK
Signing APK
Patched APK
```

### Additional context

I understand that the version constraint was removed for Reddit so that people can patch versions newer than 2024.17.0, but this version is still important because it is the latest version that supports the `Hide navigation buttons` patch.

### Acknowledgements

- [X] This issue does not reproduce on unpatched YouTube or YT Music.
- [X] This issue is not a duplicate of an existing bug report.
- [X] I did not use any settings marked as `Experimental Flags`.
- [X] I did not use `pre-release` (dev branch).
- [X] I have patched the APK according to the [documentation](https://github.com/inotia00/revanced-documentation#readme).
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by inotia00** [comment]
reflected in [revanced-patches-5.1.2](https://github.com/inotia00/revanced-patches/releases/tag/v5.1.2)

**Comment by KobeW50** [comment]
@inotia00 The patch still fails, but for a different reason:

- Device Info
RVX Manager: 1.23.5
Model: SM-S906U1
Android version: 14
Supported architectures: arm64-v8a, armeabi-v7a, armeabi
Root permissions: No

- Patch Info
App: com.reddit.frontpage v2024.17.0 (Suggested: Any)
Patches version: v5.1.2
Patches added: Hide navigation buttons
Patches removed: None
Default patch options changed: None

- Settings
Allow changing patch selection: true
Version compatibility check: true
Show universal patches: true
Remove unused library: true
Use pre-release: false
Patches source: inotia00/revanced-patches

- Logs
Reading APK
Decoding app manifest
Loading patches
Deleting existing temporary files directory
Decoding resources
Initializing lookup maps
Executing patches
Applied 10 patches
Hide ads failed: app.revanced.patcher.patch.PatchException: The patch "Hide ads" depends on "BytecodePatch (commentAdsPatch)", which raised an exception:
app.revanced.patcher.patch.PatchException: Collection contains no element matching the predicate.
	at app.revanced.patcher.Patcher$invoke$1.invokeSuspend$execute(SourceFile:90)
	at app.revanced.patcher.Patcher$invoke$1.invokeSuspend$execute(SourceFile:72)
	at app.revanced.patcher.Patcher$invoke$1.invokeSuspend(SourceFile:109)
	at app.revanced.patcher.Patcher$invoke$1.invoke(SourceFile:0)
	at app.revanced.patcher.Patcher$invoke$1.invoke(SourceFile:0)
	at kotlinx.coroutines.flow.SafeFlow.collectSafely(SourceFile:57)
	at kotlinx.coroutines.flow.AbstractFlow.collect(SourceFile:226)
	at app.revanced.manager.flutter.MainActivity$runPatcher$1$patcherResult$1$1.invokeSuspend(SourceFile:323)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(SourceFile:33)
	at kotlinx.coroutines.DispatchedTask.run(SourceFile:101)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(SourceFile:263)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(SourceFile:95)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(SourceFile:69)
	at kotlinx.coroutines.BuildersKt.runBlocking(SourceFile:1)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(SourceFile:47)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(SourceFile:1)
	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda$34(SourceFile:317)
	at app.revanced.manager.flutter.MainActivity.$r8$lambda$Oku38Rh6VOg9CiulxEElyw3q4GI(SourceFile:0)
	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda0.run(SourceFile:0)
	at java.lang.Thread.run(Thread.java:1012)
Caused by: java.util.NoSuchElementException: Collection contains no element matching the predicate.
	at app.revanced.util.BytecodeUtilsKt.findMethodOrThrow(BytecodeUtils.kt:800)
	at app.revanced.util.BytecodeUtilsKt.getWalkerMethod(BytecodeUtils.kt:532)
	at app.revanced.util.BytecodeUtilsKt.getWalkerMethod(BytecodeUtils.kt:527)
	at app.revanced.patches.reddit.ad.AdsPatchKt.commentAdsPatch$lambda$8$lambda$7(AdsPatch.kt:62)
	at app.revanced.patches.reddit.ad.AdsPatchKt.$r8$lambda$c5DYZcOtGPl6_hiYPpk8VDsgIuk(Unknown Source:0)
	at app.revanced.patches.reddit.ad.AdsPatchKt$$ExternalSyntheticLambda5.invoke(Unknown Source:2)
	at app.revanced.patcher.patch.Patch.execute(SourceFile:71)
	at app.revanced.patcher.patch.BytecodePatch.execute$revanced_patcher(SourceFile:166)
	at app.revanced.patcher.Patcher$invoke$1.invokeSuspend$execute(SourceFile:84)
	... 19 more

	at app.revanced.patcher.Patcher$invoke$1.invokeSuspend$execute(SourceFile:75)
	at app.revanced.patcher.Patcher$invoke$1.invokeSuspend(SourceFile:109)
	at app.revanced.patcher.Patcher$invoke$1.invoke(SourceFile:0)
	at app.revanced.patcher.Patcher$invoke$1.invoke(SourceFile:0)
	at kotlinx.coroutines.flow.SafeFlow.collectSafely(SourceFile:57)
	at kotlinx.coroutines.flow.AbstractFlow.collect(SourceFile:226)
	at app.revanced.manager.flutter.MainActivity$runPatcher$1$patcherResult$1$1.invokeSuspend(SourceFile:323)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(SourceFile:33)
	at kotlinx.coroutines.DispatchedTask.run(SourceFile:101)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(SourceFile:263)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(SourceFile:95)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(SourceFile:69)
	at kotlinx.coroutines.BuildersKt.runBlocking(SourceFile:1)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(SourceFile:47)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(SourceFile:1)
	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda$34(SourceFile:317)
	at app.revanced.manager.flutter.MainActivity.$r8$lambda$Oku38Rh6VOg9CiulxEElyw3q4GI(SourceFile:0)
	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda0.run(SourceFile:0)
	at java.lang.Thread.run(Thread.java:1012)
"Hide navigation buttons" patch is not supported in this version. Use Reddit 2024.17.0 or earlier.
Compiling modified resources
Ripping selected native libs
Aligning APK
Signing APK
Patched APK

**Comment by KobeW50** [comment]
Interestingly, the first two times I tested 2024.17.0 with patches 5.1.2 I got this error (and also the warning for the `Hide navigation buttons` patch appeared).

Then I patched without the `Hide ads` patch but with the `Hide navigation buttons` patch and it worked fine.

Now I am no longer able to reproduce this patching error for the `Hide ads` patch, even when I use the same patch selection as in the log from my previous comment.

Idk. Maybe it should stay closed until i or someone else is able to reproduce the new issue again.
